### PR TITLE
fix: convert SerilizedDocument to real dom

### PR DESCRIPTION
**Pull request by kilatib**
https://oatsa.slack.com/archives/C5V897QNM/p1663224082625449

### CODE REVIEW

File: models/classes/cat/CatUtils.php [hunk 1]
```diff
@@ -39,8 +38,9 @@ public static function getCatInfo(AssessmentTest $test, $namespace = '')
         $info = [];
         
         foreach ($test->getComponentsByClassName('assessmentSection') as $assessmentSection) {
-            if (($selection = $assessmentSection->getSelection()) !== null && (($xmlExtension = $selection->getXml())) !== null) {
-                $xpath = new \DOMXPath($xmlExtension);
+            $xmlExtension = new DOMDocument();
+            if (($selection = $assessmentSection->getSelection()) !== null && $xmlExtension->loadXML($selection->getXml()->saveXML())) {
+                $xpath = new \DOMXPath($xmlExtension->);
```
Comment (poyuki) [inline models/classes/cat/CatUtils.php:43]:
```suggestion
                $xpath = new \DOMXPath($xmlExtension);
```

**Comment by kilatib** [comment]
Test down with other problem about number rounding

**Review by uncleempty — COMMENTED** [review]
Stupid question: since in any case `$selection->getXml()` will cause a load into `SerializableDomDocument` (ofc if `$this->xml` is `null`) from `$this->getXmlString()` - doesn't it make more sense to just load the XML via `$selection->getXmlString()`?

File: models/classes/cat/CatUtils.php [hunk 2]
```diff
@@ -39,7 +38,8 @@ public static function getCatInfo(AssessmentTest $test, $namespace = '')
         $info = [];
         
         foreach ($test->getComponentsByClassName('assessmentSection') as $assessmentSection) {
-            if (($selection = $assessmentSection->getSelection()) !== null && (($xmlExtension = $selection->getXml())) !== null) {
+            $xmlExtension = new DOMDocument();
```
Comment (poyuki) [inline models/classes/cat/CatUtils.php:41]:
```suggestion
            $xmlExtension = new \DOMDocument();
```

File: models/classes/cat/CatUtils.php [hunk 3]
```diff
@@ -94,8 +94,8 @@ public static function isAssessmentSectionAdaptive(AssessmentSection $section, $
         }
         
         $isAdaptive = false;
-        
-        if (($selection = $section->getSelection()) !== null && (($xmlExtension = $selection->getXml())) !== null) {
+        $xmlExtension = new DOMDocument();
```
Comment (poyuki) [inline models/classes/cat/CatUtils.php:97]: could you please apply or import it
```suggestion
        $xmlExtension = new \DOMDocument();
```

**Comment by poyuki** [comment]
still error provided, it's related to my prev comments 

> 1) oat\taoQtiTest\test\unit\cat\CatUtilsTest::testGetCatInfo with data set #0 ('/home/runner/work/extension-t...on.xml', array(array('http://www.my-cat-engine.com/api/', 'cat/settings.data', 'cat/usagedata.data', 'cat/metadata.data')))
Error: Class 'oat\taoQtiTest\models\cat\DOMDocument' not found

**Review by poyuki — APPROVED** [review]

**Review by wazelin — APPROVED** [review]

**Pull request merged**